online-shop/src/app/product-details/product-details.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Output, EventEmitter } from '@angular/core';
+import { PRODUCTS } from '../products';
+import { Product } from '../product';
+
+@Component({
+  selector: 'app-product-details',
+  templateUrl: './product-details.component.html',
+  styleUrls: ['./product-details.component.css']
+})
+
+export class ProductDetailsComponent implements OnInit {
+
+  product;
+
+  @Output() p =  new EventEmitter<Product>(); 
```

Comment:
You should use better naming for your outputs. `p` does not convey any information... `addToCart` is better...